MathLib/LinAlg/LinearSolver.cpp

```diff
@@ -0,0 +1,105 @@
+#include "LinearSolver.h"
+
+namespace
+{
+//! Describes the default linear solver type to be used together with the given
+//! \c Matrix and \c Vector types.
+template<typename Matrix, typename Vector>
+struct DefaultLinearSolver;
+}
+
+namespace MathLib
+{
+
+template<typename Matrix, typename Vector, typename Solver>
+std::unique_ptr<LinearSolver<Matrix, Vector> >
+createLinearSolver(BaseLib::ConfigTree const*const config)
+{
+    using AbstractLS = LinearSolver<Matrix, Vector>;
+    return std::unique_ptr<AbstractLS>{new Solver{"", config}};
+}
+
+template<typename Matrix, typename Vector>
+std::unique_ptr<LinearSolver<Matrix, Vector> >
+createLinearSolver(BaseLib::ConfigTree const*const config)
+{
+    using Solver = typename DefaultLinearSolver<Matrix, Vector>::Solver;
+    return createLinearSolver<Matrix, Vector, Solver>(config);
+}
+
+} // MathLib
+
+
+// specializes the DefaultLinearSolver<> class
+#define DfltLinSlv(MAT, VEC, SLV) \
```

Comment:
I'm also suggesting to write out all of the "short cuts". The time to recognize DfltLinSlv is (for me) definitely longer than for DefaultLinearSolver.

Macros should be upper case.